### PR TITLE
Make identifier quote character for packName() customizable

### PR DESCRIPTION
**Pull request by na-gupta**
MySQL uses backquotes instead of double quotes to quote identifiers with spaces or special characters. This change makes this character customizable using QuerySubstitutionNames keeping the default the same as it currently is hardcoded to (double quotes).

### CODE REVIEW

File: Sources/SwiftKuery/AggregateColumnExpression.swift [hunk 1]
```diff
@@ -37,7 +37,7 @@ public struct AggregateColumnExpression: Field {
     public func build(queryBuilder: QueryBuilder) throws -> String {
```
Comment (irar2) [inline Sources/SwiftKuery/AggregateColumnExpression.swift]: Please add 2017 to the copyright here and in other files.

File: Sources/SwiftKuery/QueryBuilder.swift [hunk 2]
```diff
@@ -43,6 +43,8 @@ public class QueryBuilder {
         case booleanFalse
         /// The ALL function for subqueries.
         case all
+        /// character used to quote identifiers (table name,column name, etc) with spaces or special characters
+        case identifierQuoteCharacter
```
Comment (irar2) [inline Sources/SwiftKuery/QueryBuilder.swift]: Please change to: 
"Character used to quote identifiers (table name, column name, etc.) with spaces or special characters."
(start sentence with uppercase, spaces, periods).

File: Sources/SwiftKuery/Utils.swift [hunk 3]
```diff
@@ -39,10 +39,11 @@ func packType(_ item: Any, queryBuilder: QueryBuilder) throws -> String {
     }
 }
 
-func packName(_ name: String) -> String {
+func packName(_ name: String, queryBuilder: QueryBuilder) -> String {
     var result = name
-    if result.contains(" ") && !result.hasPrefix("\"") {
-        result = "\"" + result + "\""
+    let identQuoteChar = queryBuilder.substitutions[QueryBuilder.QuerySubstitutionNames.identifierQuoteCharacter.rawValue]
```
Comment (irar2) [inline Sources/SwiftKuery/Utils.swift]: Please use full words in variable names, i.e. identifierQuoteCharacter instead of identQuoteChar.

File: Sources/SwiftKuery/Table.swift [hunk 4]
```diff
@@ -27,10 +27,7 @@ open class Table: Buildable {
     /// The name of the table to be used inside a query, i.e., either its alias (if exists)
     /// or its name.
     public var nameInQuery: String {
-        if let alias = alias {
-            return packName(alias)
-        }
-        return packName(_name)
+        return alias ?? _name
```
Comment (irar2) [inline Sources/SwiftKuery/Table.swift]: Why did you remove the call to packName?

**Comment by na-gupta** [comment]
Addressed review. Thanks!
